include/NovelRT/Audio/AudioService.h

```diff
@@ -11,51 +11,43 @@
 namespace NovelRT::Audio {
   class AudioService {
   private:
-    int _nextChannel;
-    int _musicTime;
-    int _musicPausedTime;
-    const int NOVEL_MIXER_CHANNELS = 2;
-    const int MIXER_TRUE = 1;
-    const int MIXER_INFINITE_LOOP = -1;
-    const int MIXER_NO_LOOP = 0;
-    const int MIXER_NO_EXPLICIT_CHANNEL = -1;
-    const int NOVEL_MUSIC_CHANNEL = -3;
+    const ALfloat _pitch = 1.0f;
+    
+    LoggingService _logger;
+    ALint _musicLoopAmount;
+    ALuint _musicSource;
+    ALint _musicSourceState;
+    ALint _soundLoopAmount;
+    ALuint _soundSource;
+    ALint _soundSourceState;
     SoundBank _sounds;
     MusicBank _music;
-    ChannelMap _channelMap;
-    LoggingService _logger;
+    std::string _deviceName;
+    Utilities::Lazy<std::unique_ptr<ALCdevice, void(*)(ALCdevice*)>> _device;
+    Utilities::Lazy<std::unique_ptr<ALCcontext, void(*)(ALCcontext*)>> _context;
+
+    ALuint readFile(std::string input);
+    std::string getALError();
 
-    int convertToMixVolume(float value);
-    std::string findByChannelMap(int channel);
-    std::string getSDLError();
-    void incrementNextChannel();
-    void logIfSDLFailure(std::function<int(Uint32)> sdlFunction, Uint32 sdlFlag);
-    void logIfMixerFailure(std::function<int(int)> mixerFunction, int mixerFlag);
-    void logIfMixerFailure(std::function<int(int,Uint16,int,int)> mixerFunction, int freq, Uint16 mixerFormat, int channels, int sampleSize);
   public:
-    bool isInitialized;
+    bool isInitialised;
 
     AudioService();
     ~AudioService();
 
     bool initializeAudio();
-    void load(std::string input, bool isMusic);
-    void unload(std::string input, bool isMusic);
-    void playSound(std::string soundName, int loops);
-    void stopSound(std::string soundName);
-    void setSoundVolume(std::string soundName, float val);
-    void setSoundPosition(std::string soundName, int angle, int distance);
-    void setSoundDistance(std::string soundName, int distance);
-    void setSoundPanning(std::string soundName, int leftChannel, int rightChannel);
+    std::vector<ALuint>::iterator load(std::string input, bool isMusic);
```

Comment:
Similar note for audio branch...

We probably don't want to tie ourselves to `ALuint` here in the public API.